### PR TITLE
Added the possibility of selecting the source to display.

### PR DESCRIPTION
**Pull request by Eiyeron**
Something to display the list of available sources might be interesting.

Beware that I slightly changed the way to stop the source search as it didn't show my last source. Make sure it doesn't crash on your side either.

### CODE REVIEW

**Comment by HaCk3Dq** [comment]
Okay, thnx
Will test it

**Comment by patrick96** [comment]
@HaCk3Dq did this work for you? With the default configuration I'm getting

```
Read : 1
Setup : 1
0->alsa_output.pci-0000_00_1b.0.analog-stereo.monitor
[1]    24017 segmentation fault (core dumped)  ./spectrumyzer.py
```

Gonna investigate further

**Comment by HaCk3Dq** [comment]
U should setup the source to be "0", if u change it to "1" it will crash (i've had the same crap)

**Comment by HaCk3Dq** [comment]
@Eiyeron Can u catch this segfault and close the app with `Error("wrong source selected")`?

File: src/impulse.c [hunk 1]
```diff
@@ -58,15 +59,18 @@ static void quit( int ret ) {
...
 
-	if ( is_last || device != NULL )
+	if(swept_through)
 		return;
+	if ( is_last || device != NULL )
+		swept_through = 1;
+
+	printf("%d->%s\n",i->index, i->name);
 
 	assert(i);
```
Comment (patrick96) [inline src/impulse.c:69]: I have the same problem with both `0` and `1`
This is the line that crashes, I have managed to fix it by changing `if(swept_through)` with `if(swept_through || !i)`
I will open a PR shortly, if you think there is a better way to solve it, please tell me

Comment (HaCk3Dq) [reply src/impulse.c:69]: Seems okay, but would be much better to catch the wrong source selected

**Pull request merged**